### ISSUE TITLE
[pyclustering.cluster.kmeans] USER_DEFINED metric does not work

### ISSUE THREAD

**Issue by annoviko** [annoviko/pyclustering#471]
> 
> Here are several code examples how to create distance metric:
> 
> ```python
> metric = distance_metric(type_metric.EUCLIDEAN);
> distance = metric([1.0, 2.5], [-1.2, 3.4]);
> ```
> 
> ```python
> user_function = lambda point1, point2: point1[0] + point2[0] + 2;
> metric = distance_metric(type_metric.USER_DEFINED, func=user_function);
> distance = metric([2.0, 3.0], [1.0, 3.0]);
> ```
> 
> And here is an example how to use metric in K-Medoids:
> 
> ```python
> from pyclustering.utils.metric import type_metric, distance_metric;
> 
> # create Minkowski distance metric with degree equals to '2'
> metric = distance_metric(type_metric.MINKOWSKI, degree=2);
> 
> # create K-Medoids algorithm with specific distance metric
> kmedoids_instance = kmedoids(sample, initial_medoids, metric=metric);
> 
> # run cluster analysis and obtain results
> kmedoids_instance.process();
> clusters = kmedoids_instance.get_clusters();
> ```
> 
> Changes are available on branch '0.8.dev' and will be available in the next pyclustering release - 0.8.1.

that seems to work just for kmedoids method, 

`user_function = lambda point1, point2: point1[0] + point2[0] + 2
metric = distance_metric(type_metric.USER_DEFINED, func=user_function)
sample = read_sample(SIMPLE_SAMPLES.SAMPLE_SIMPLE3)
kmedoids_instance = kmedoids(data=sample, initial_index_medoids=[2,3], metric=metric)

kmedoids_instance.process()
clusters = kmedoids_instance.get_clusters()`

but **fail to run kmeans**, like that:

`user_function = lambda point1, point2: point1[0] + point2[0] + 2
metric = distance_metric(type_metric.USER_DEFINED, func=user_function)
sample = read_sample(SIMPLE_SAMPLES.SAMPLE_SIMPLE3)
start_centers = [[0.2, 0.1], [4.0, 1.0]]
kmeans_instance = kmeans(data=sample,
                            initial_centers = start_centers,
                            metric=metric)

kmeans_instance.process()
clusters = kmeans_instance.get_clusters()`
i'm new to pyclustering. Thanks for ur help.

_Originally posted by @xunhs in https://github.com/annoviko/pyclustering/issues/417#issuecomment-436551478_

**Comment by annoviko** [comment]
The problem has been corrected on the branch 0.8.dev:
```bash
mkdir pyclustering
cd pyclustering
git clone https://github.com/annoviko/pyclustering.git . -b 0.8.dev
```

0.8.dev contains changes related 0.8.2 - and it is available yet, so core should be rebuilt:
```bash
cd ccore
make <your target>
```

Instruction how to build core: https://github.com/annoviko/pyclustering/wiki/Core-of-the-PyClustering

**Comment by xunhs** [comment]
> The problem has been corrected on the branch 0.8.dev:
> 
> ```shell
> mkdir pyclustering
> cd pyclustering
> git clone https://github.com/annoviko/pyclustering.git . -b 0.8.dev
> ```
> 
> 0.8.dev contains changes related 0.8.2 - and it is available yet, so core should be rebuilt:
> 
> ```shell
> cd ccore
> make <your target>
> ```
> 
> Instruction how to build core: https://github.com/annoviko/pyclustering/wiki/Core-of-the-PyClustering
> 
> The correction will be available in release 0.8.2.

Thanks very much!
